src/core/patterns.js

```diff
@@ -152,3 +152,78 @@ foam.CLASS({
     }
   ]
 });
+
+/** Flyweight classes create instances that can create flyweight copies.
```

Comment:
This is not what a "Flyweight" is.  You could call it a "lightweight" or something else, but Flyweight has a specific meaning, which is different from this.